### PR TITLE
[FA] 12-2.md Translation

### PR DESCRIPTION
**Pull request by thisistayeb**

### CODE REVIEW

File: docs/fa/week12/12-1.md [hunk 1]
```diff
@@ -0,0 +1,406 @@
...
+--->
+یک مزیت بزرگ در مورد توجه چند سر این است که به صورت موازی بسیار قابل محاسبه است. برخلاف RNNها، که سرهای ماژول های توجه چند سر و همه مراحل گام را به طور همزمان محاسبه می کند. یک مشکل محاسبه همزمان همه مراحل زمانی این است که می تواند کلمات آینده را نیز بررسی کند، در حالی که ما فقط می خواهیم به کلمات قبلی شرط بگذاریم. یک راه حل برای آن چیزی است که ** پوشش خود-توجه ای ** نامیده می شود. پوشش یک ماتریس مثلثی فوقانی است که در مثلث پایین صفر و در مثلث بالایی بی نهایت منفی دارد. تأثیر افزودن این پوشش به خروجی ماژول توجه این است که هر کلمه به سمت چپ دارای نمره توجه بسیار بیشتری نسبت به کلمات به سمت راست است، بنابراین مدل در عمل فقط بر روی کلمات قبلی تمرکز دارد. استفاده از پوشش در مدل زبان بسیار مهم است زیرا آن را از نظر ریاضی صحیح می کند، با این حال، در رمزگذارهای متن، متن دو زبانه می تواند مفید باشد.
```
Comment (telegraphroad) [inline docs/fa/week12/12-1.md:217]: I'd suggest something a bit more fluent like
یک مزیت بزرگ در مورد توجه چندسر این است که محاسبات آن را به راحتی می‌توان به صورت موازی انجام داد.
I'm not sure about translating time-steps into گام. maybe گام‌های زمانی? sounds awful but conveys the sense that the steps are through time, not on some other axis
I'm not sure about translating mask to پوشش, might be confused with "covering". Does it conform throughout the translations for other parts of the course? If not leaving the "mask" as is might be a good idea.

**Review by telegraphroad — CHANGES_REQUESTED** [review]
Very nice translation, high quality, Farsi is fluent and pleasant to read. Thanks!

I did some spot checks and found some teeny tiny things that might need your attention. I left comments on one of the paragraphs to give you an idea about what I'm talking about.

My gentle suggestion would be to do a quick review of the translation, maybe rephrase some of the sentences for the ease of read, and pay special attention to the words that will be used in different contexts. For example and speaking of the devil, I saw you translate "context" as متن in one of the paragraphs. It conveys the meaning adequately in _that sentence_, but might be misleading overall because "context" has a specific definition and place in the whole "deep learning latent space" :)

File: docs/fa/week12/12-1.md [hunk 2]
```diff
@@ -0,0 +1,406 @@
...
+</figure>
+<!---
+**Why the model is so good:**
+
+1. It gives direct connections between each pair of words. Each word can directly access the hidden states of the previous words, mitigating vanishing gradients. It learns very expensive function very easily
+2. All time-steps are computed in parallel
+3. Self-attention is quadratic (all time-steps can attend to all others), limiting maximum sequence length
+--->
+
+**چرا این مدل خوب است؟**
```
Comment (telegraphroad) [inline docs/fa/week12/12-1.md:242]: Not to nitpick, but چرا این مدل این‌قدر خوب است might convey the extra emphasis here

File: docs/fa/week12/12-1.md [hunk 3]
```diff
@@ -0,0 +1,406 @@
...
+۱. ارتباط مستقیمی بین هر جفت کلمه ایجاد می کند. هر کلمه می تواند مستقیماً به حالت های پنهان کلمات قبلی دسترسی پیدا کند و گرادیان های ناپدید شده را کاهش دهد. تابع پرهزینه ای را به راحتی یاد می‌گیرد.
```
Comment (telegraphroad) [inline docs/fa/week12/12-1.md:243]: Wouldn't مشکل گرادیان‌های ناپدیدشونده را برطرف می‌کند make a bit more sense than کاهش می‌دهد? I mean they are already vanishing, right? :)

File: docs/fa/week12/12-1.md [hunk 4]
```diff
@@ -0,0 +1,406 @@
...
+## [برخی از ترفندها (به ویژه برای توجه چند سر و رمزگذاری موقعیتی) و رمزگشایی مدل های زبان](https://www.youtube.com/watch?v=6D4EWKJgNn0&t=1975s)
+
+### ترفند ۱: استفاده گسترده از نرمال سازی لایه برای تثبیت آموزش واقعاً مفید است
+
+- برای دگرگون ساز بسیار مهم است
+ 
+ 
+### ترفند ۲: دست گرمی + برنامه زمانی آموزش ریشه مربع معکوس
+
+- از برنامه زمانی نرخ یادگیری استفاده کنید: برای اینکه دگرگون سازها به خوبی کار کنند، باید سرعت یادگیری خود را از صفر تا هزار مرحله به صورت خطی کاهش دهید
```
Comment (telegraphroad) [inline docs/fa/week12/12-1.md:289]: I'd suggest using زمان‌بندی instead of برنامه‌ی زمانی

File: docs/fa/week12/12-1.md [hunk 5]
```diff
@@ -0,0 +1,406 @@
...
+---> 
+## مدل های زبان رمزگشا
+
+اکنون می توانیم توزیع احتمال را روی متن آموزش دهیم - در حال حاضر اساساً می توانیم به صورت نمایی بسیاری از خروجی های ممکن را بدست آوریم، بنابراین نمی توانیم حداکثر را محاسبه کنیم. هر انتخابی که برای اولین کلمه خود انجام می دهید می تواند بر سایر تصمیمات تأثیر بگذارد.
+بنابراین، با توجه به این، رمزگشایی حریصانه به شرح زیر معرفی شد.
+
+
+### رمزگشایی حریص کار نمی کند
+
+ما محتمل ترین کلمه را در هر مرحله زمان می گیریم. با این وجود، هیچ تضمینی این محتمل ترین توالی ممکن باشد، زیرا اگر مجبور باشید در مرحله ای آن مرحله را انجام دهید، دیگر هیچ راهی برای پیگیری جستجوی خود برای پس‌گرد سایر نشست‌های قبلی ندارید.
```
Comment (telegraphroad) [inline docs/fa/week12/12-1.md:372]: هیچ تضمینی برای این‌که این محتمل‌ترین توالی ممکن باشد وجود ندارد

**Review by Atcold — APPROVED** [review]
Merging for now. If a review is necessary, please open a new PR.
